src/EnergyPlus/InputProcessing/InputProcessor.cc

```diff
@@ -453,6 +453,30 @@ bool InputProcessor::findDefault(Real64 &default_value, json const &schema_field
     return false;
 }
 
+bool InputProcessor::getDefaultRealValue(std::string const &objectWord, std::string const &fieldName, Real64 &value)
+{
+    auto find_iterators = objectCacheMap.find(objectWord);
+    if (find_iterators == objectCacheMap.end()) {
+        auto const tmp_umit = caseInsensitiveObjectMap.find(convertToUpper(objectWord));
+        if (tmp_umit == caseInsensitiveObjectMap.end() || epJSON.find(tmp_umit->second) == epJSON.end()) {
+            return false;
+        }
+        find_iterators = objectCacheMap.find(tmp_umit->second);
+    }
+    auto const &epJSON_schema_it = find_iterators->second.schemaIterator;
+    auto const &epJSON_schema_it_val = epJSON_schema_it.value();
+    auto const &schema_obj_props = epJSON_schema_it_val["patternProperties"][".*"]["properties"];
+    auto const &sizing_factor_schema_field_obj = schema_obj_props.at(fieldName);
+    Real64 defaultValue = 0.0;
+    bool defaultFound = findDefault(defaultValue, sizing_factor_schema_field_obj);
+    if (defaultFound) {
+        value = defaultValue;
+        return true;
+    } else {
+        return false;
+    }
+}
```

Comment:
@mbadams5 I cleaned it up a bit, now it returns a bool flag and passes the value back by reference.  Seems to work fine, but I'm open to improvements/comments.

Reply:
@Myoldmopar I would rename it `getDefaultValue` or something else like that. You don't need the `Real` part of the name. I would then create another nearly identical function where you pass in a string so the function is overloaded just like `findDefault`. Also, I would pass the `value` input parameter to `findDefault`. The `findDefault` behavior is the correct one with regards to existing code.

```c++
    bool defaultFound = findDefault(value, sizing_factor_schema_field_obj);
    return defaultFound;
```
if it complains about not being about to use lvalue or something else then:
```c++
    Real64 defaultValue = 0.0;
    bool defaultFound = findDefault(defaultValue, sizing_factor_schema_field_obj);
    value = defaultValue;
    return defaultFound;
```

Reply:
Thanks @mbadams5 I'll do it!